shell/platform/windows/accessibility_root_node.cc

```diff
@@ -0,0 +1,287 @@
+                                                  VARIANT start,
+                                                  VARIANT* end) {
+  IAccessible* target;
+  if ((target = GetTargetAndChildID(&start))) {
+    return target->accNavigate(nav_dir, start, end);
+  }
+  return S_FALSE;
+}
+
+IFACEMETHODIMP AccessibilityRootNode::get_accChild(VARIANT var_child,
+                                                   IDispatch** disp_child) {
+  if (V_VT(&var_child) != VT_I4) {
+    return E_FAIL;
+  }
+  LONG child_id = V_I4(&var_child);
+  if (child_id == CHILDID_SELF) {
+    *disp_child = this;
+  } else if (!window_accessible_) {
+    return E_FAIL;
+  } else if (child_id == kWindowChildId) {
+    *disp_child = window_accessible_;
+  } else if (child_id == kAlertChildId && alert_accessible_) {
+    *disp_child = alert_accessible_;
+  } else if (child_id < 0) {
+    // A negative child ID can be used to refer to an AX node directly by its
+    // ID.
+    return window_accessible_->get_accChild(var_child, disp_child);
+  } else {
+    return E_FAIL;
+  }
+  (*disp_child)->AddRef();
+  return S_OK;
+}
+
+IFACEMETHODIMP AccessibilityRootNode::get_accChildCount(LONG* child_count) {
+  LONG children = 0;
+  if (window_accessible_) {
+    children++;
+  }
+  if (alert_accessible_) {
+    children++;
+  }
+  *child_count = children;
+  return S_OK;
+}
+
+IFACEMETHODIMP AccessibilityRootNode::get_accDefaultAction(VARIANT var_id,
+                                                           BSTR* def_action) {
+  IAccessible* target;
+  if ((target = GetTargetAndChildID(&var_id))) {
+    return target->get_accDefaultAction(var_id, def_action);
+  }
+  *def_action = nullptr;
+  return S_FALSE;
+}
+
+IFACEMETHODIMP AccessibilityRootNode::get_accDescription(VARIANT var_id,
+                                                         BSTR* desc) {
+  IAccessible* target;
+  if ((target = GetTargetAndChildID(&var_id))) {
+    return target->get_accDescription(var_id, desc);
+  }
+  return E_FAIL;
+}
+
+IFACEMETHODIMP AccessibilityRootNode::get_accFocus(VARIANT* focus_child) {
+  if (window_accessible_) {
+    return window_accessible_->get_accFocus(focus_child);
+  }
+  focus_child->vt = VT_EMPTY;
+  return S_OK;
+}
+
+IFACEMETHODIMP AccessibilityRootNode::get_accKeyboardShortcut(VARIANT var_id,
+                                                              BSTR* acc_key) {
+  IAccessible* target;
+  if ((target = GetTargetAndChildID(&var_id))) {
+    return target->get_accKeyboardShortcut(var_id, acc_key);
+  }
+  return E_FAIL;
+}
+
+IFACEMETHODIMP AccessibilityRootNode::get_accName(VARIANT var_id,
+                                                  BSTR* name_bstr) {
+  if (V_I4(&var_id) == CHILDID_SELF) {
+    std::wstring name = L"ROOT_NODE_VIEW";
+    *name_bstr = SysAllocString(name.c_str());
+    return S_OK;
+  }
+  IAccessible* target;
+  if ((target = GetTargetAndChildID(&var_id))) {
+    return window_accessible_->get_accName(var_id, name_bstr);
```

Comment:
Why hard code to window_accessible_? if this is intended you can get rid of the `IAccessible* target` i think. Also add some comment explains why